autoload/go/cmd.vim

```diff
@@ -136,7 +136,7 @@ endfunction
 " suitable for long running apps, because vim is blocking by default and
 " calling long running apps will block the whole UI.
 function! go#cmd#Run(bang, ...) abort
-  if has('nvim')
+  if has('nvim') || has('terminal')
```

Comment:
@bhcleek 
Believe this can go away in favor of just checking:
```
if go#config#TermEnabled()
```

Reply:
I agree